packages/actionmenu/src/react/index.js

```diff
@@ -24,33 +24,148 @@ const styles = {
+      this.navigate('up', evt)
+    } else if (evt.key === 'Tab') {
+      this.navigateTab(evt)
+    }
+  }
+  handleDividerFocus() {
+    if (this.state.activeDirection === 'down') {
+      const newIndex = this.state.activeIndex + 1
+      const itemsCount = React.Children.count(this.props.children)
+      const activeIndex = newIndex > itemsCount - 1 ? itemsCount - 1 : newIndex
+      this.setState({ activeIndex, activeDirection: 'down' })
+    } else if (this.state.activeDirection === 'up') {
+      const newIndex = this.state.activeIndex - 1
+      const activeIndex = newIndex <= 0 ? 0 : newIndex
+      this.setState({ activeIndex, activeDirection: 'up' })
+    }
+  }
+  focusItemAtIndex(i) {
+    this.setState({ activeIndex: i })
+  }
+  focusItemAtIndexWithMouse(i) {
+    this.focusItemAtIndex(i)
+    this.setState({ isKeyboarding: false })
+  }
+  navigateOut(evt) {
+    evt.stopPropagation()
+    evt.preventDefault()
+    if (typeof this.props.onClose === 'function') this.props.onClose()
+  }
+  navigate(direction, evt) {
+    evt.stopPropagation()
+    evt.preventDefault()
+    const activeIndex = calcNextIndex(
+      React.Children.map(this.props.children, c => c.props),
+      direction === 'down' ? 1 : -1,
+      this.state.activeIndex
+    )
+
+    this.setState({
+      activeIndex,
+      activeDirection: direction,
+      isKeyboarding: true
+    })
+  }
+  navigateTab(evt) {
+    const direction = evt.shiftKey ? 'up' : 'down'
+    const { activeIndex } = this.state
+    const lastIndex = React.Children.count(this.props.children) - 1
+    const atEdge =
+      (direction === 'up' && activeIndex === 0) ||
+      (direction === 'down' && activeIndex === lastIndex)
+
+    if (atEdge) {
+      this.navigateOut(evt)
+    } else {
+      this.navigate(direction, evt)
+    }
+  }
+  render() {
+    const { ref: innerRef, ...rest } = this.props
+
+    return (
+      <Menu
+        {...rest}
+        innerRef={innerRef}
+        onKeyDown={this.handleKeyDown}
+        role="menu"
+      >
+        {React.Children.map(this.props.children, (child, i) =>
+          React.cloneElement(child, {
+            isActive: i === this.state.activeIndex,
+            shouldFocusOnMount: this.props.shouldFocusOnMount,
+            _i: i,
+            _isKeyboarding: this.state.isKeyboarding,
+            _onItemFocus: this.focusItemAtIndex,
+            _onDividerFocus: this.handleDividerFocus,
+            _onMouseOver: this.focusItemAtIndexWithMouse,
+            _origin: this.props.origin
+          })
+        )}
+      </Menu>
+    )
+  }
+}
+ActionMenu.displayName = 'ActionMenu'
+
 const ItemIcon = props => {
   return (
     <div {...glamor.css(css['.psds-actionmenu__item__icon'])}>
-      {React.cloneElement(props.children, { size: Icon.sizes.medium })}
+      {React.cloneElement(props.children, { size: iconVars.sizes.medium })}
     </div>
   )
 }
+ItemIcon.propTypes = {
+  children: PropTypes.element // Icon
```

Comment:
even though was doing elementOfType on stuff, this was a bundle size compromise.